test/vol.c

```diff
@@ -162,6 +295,85 @@ static const H5VL_class_t fake_vol_g = {
     NULL /* optional     */
 };
 
+/*-------------------------------------------------------------------------
```

Comment:
Does this test code demonstrate/exercise the API as it might be used by in a real use case or is it really just unit-checking the API? If the latter, I would want to see a real and well-documented end-user example which exercises key (if not all) modalities of use (or separate tests for separate modalities) also with this code base to understand how the user-facing API is intended to be used.

Reply:
This is a unit test and is only checking the API for correctness.   There’s a separate VOL connector testing suite under development, here:   https://github.com/HDFGroup/vol-tests